### PR TITLE
#14235: adding dialect based on connection type to LineageParser

### PR DESCRIPTION
**Pull request by LucasGarcia07**
### Describe your changes:

Fixes 14235, adding dialect based on db_service_name to the LineageParser.

I worked on because I'm create lineage based on bigquery dialect and sqlfluff fails when use ansi dialect to validate.

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by LucasGarcia07** [comment]
@harshach

**Comment by harshach** [comment]
Thanks @LucasGarcia07 for the contribution.
@ulixius9 can you do a review

File: ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py [hunk 1]
```diff
@@ -275,7 +277,15 @@ def _yield_lineage_from_query(
 
         database_name = database.details.db if database and database.details else None
 
-        lineage_parser = LineageParser(query)
+        db_service = self.metadata.get_by_name(DatabaseService, db_service_name)
+
+        lineage_parser = LineageParser(
+            query,
+            ConnectionTypeDialectMapper.dialect_of(
+                db_service.connection.config.type.value
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py:285]:
```suggestion
                db_service.serviceType.value
```

as `connection` is an optional field but `serviceType` is a required field in the DatabaseService object
and will contain the same value, @LucasGarcia07 can you please validate and test is out once if this works?

Comment (ulixius9) [reply ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py:285]: also the tests are failing can you please check that as well?

Comment (LucasGarcia07) [reply ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py:285]: thanks for the review, i will fix and test!

**Review by ulixius9 — APPROVED** [review]

**Comment by ulixius9** [comment]
Thanks @LucasGarcia07 for the pr ❤️

**Pull request merged**